src/apps/voting/components/ModalFlows/NewVote/CreateNewVote.js

```diff
@@ -4,25 +4,34 @@ import { useMultiModal } from '@/components/MultiModal/MultiModalProvider'
 import RequiredTokensError from '@/components/RequiredTokensInfo'
 import { useFee } from '@/providers/Fee'
 import { TermsOfUseDisclaimer } from '@/components/Disclaimers'
+import { VOTING_REFERENCE_SEPARATOR } from '@/constants'
+import { URL_REGEX } from '@/utils/text-utils'
 
 function CreateNewVote({ getTransactions }) {
   const { hasFeeTokens } = useFee()
-  const [question, setQuestion] = useState('')
+  const [title, setTitle] = useState('')
+  const [reference, setReference] = useState('')
   const { next } = useMultiModal()
-  const disableButton = !question.length || !hasFeeTokens
+  const disableButton =
+    !title.length || !hasFeeTokens || !URL_REGEX.test(reference)
 
-  const handleQuestionChange = useCallback(event => {
-    const updatedQuestion = event.target.value
-    setQuestion(updatedQuestion)
+  const handleTitleChange = useCallback(event => {
+    const updatedTitle = event.target.value
+    setTitle(updatedTitle)
+  }, [])
+
+  const handleReferenceChange = useCallback(event => {
+    const updatedReference = event.target.value
+    setReference(updatedReference)
   }, [])
 
   const handleOnCreateVote = useCallback(() => {
     next()
 
     getTransactions(() => {
       next()
-    }, question)
-  }, [getTransactions, question, next])
+    }, title.concat(VOTING_REFERENCE_SEPARATOR, reference))
```

Comment:
Just like we have a function to parse these contexts, let's have one to build them. Let's put this logic inside a `buildContext(title, reference?)` function that concatenates using the separator if a `reference` param is passed